### ISSUE TITLE
this.mods is kept down the tree

### ISSUE THREAD

**Issue by tadatuta** [bem/bem-xjst#80]
For BEMJSON `{ block: 'b1' }` and template
```js
block('b1')(
    def()(function() {
        this.mods.m1 = 'v1';
        return applyNext();
    }),
    content()([
        {
            block: 'b2'
        }
    ])
);
```
result is
```html
<div class="b1 b1_m1_v1"><div class="b2 b2_m1_v1"></div></div>
```
but
```html
<div class="b1 b1_m1_v1"><div class="b2"></div></div>
```
expected.

**Comment by indutny** [comment]
You have to use `local`. And yeah, it is this way at the moment.

I'll do benchmarks to see if creating `mods` for each block will be much slower than what we have now.

**Comment by indutny** [comment]
Fixed in 3c7f239

**Comment by indutny** [comment]
No performance loss seen
